### PR TITLE
[lock] Add documentation for Postgresql

### LINKED ISSUE #14362: [lock] Add store dedicated to postgresql
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#38346
| PR author(s) | @jderusse
| Merged in    | 5.x

### PR DESCRIPTION
**Pull request by jderusse**
Fixes #14362

### CODE REVIEW

File: components/lock.rst [hunk 1]
```diff
@@ -427,6 +427,29 @@ You can also create this table explicitly by calling the
 :method:`Symfony\\Component\\Lock\\Store\\PdoStore::createTable` method in
 your code.
 
+.. _lock-store-pgsql:
+
+PostgreSqlStore
+~~~~~~~~~~~~~~~
+
+The PostgreSqlStore uses `Advisory Locks`_ provided by Postgresql. It requires a
+ `PDO`_ connection, a `Doctrine DBAL Connection`_, or a
```
Comment (javiereguiluz) [inline components/lock.rst]: Don't mind the `DOCtor-RST^ messages. In this case the error is that you have a white space at the beginning of this and the next lines.

File: components/lock.rst [hunk 2]
```diff
@@ -735,6 +759,20 @@ have synchronized clocks.
...
 
+PostgreSqlStore
+~~~~~~~~~~~~~~~
+
+The PdoStore relies on the `Advisory Locks`_ properties of the Postgresql
+database. That means that by using :ref:`PostgreSqlStore <lock-store-pgsql>`
+the locks will be automatically released at the end of the session in case the
+client cannot unlock for any reason.
+
+If the Postgresql service or the machine hosting it restarts, every lock would
```
Comment (javiereguiluz) [inline components/lock.rst]: `Postgresql service` -> `PostgreSQL service`

File: components/lock.rst [hunk 3]
```diff
@@ -735,6 +759,20 @@ have synchronized clocks.
...
+be lost without notifying the running processes.
+
+If the tcp connection is lost, the Postgresql may release locks without
```
Comment (javiereguiluz) [inline components/lock.rst]: `tcp` -> `TCP`  and `Postgresql` -> `PostgreSQL`

File: components/lock.rst [hunk 4]
```diff
@@ -735,6 +759,20 @@ have synchronized clocks.
     To ensure locks don't expire prematurely; the TTLs should be set with
     enough extra time to account for any clock drift between nodes.
 
+PostgreSqlStore
+~~~~~~~~~~~~~~~
+
+The PdoStore relies on the `Advisory Locks`_ properties of the Postgresql
```
Comment (javiereguiluz) [inline components/lock.rst]: `Postgresql` -> `PostgreSQL`

File: components/lock.rst [hunk 5]
```diff
@@ -427,6 +427,29 @@ You can also create this table explicitly by calling the
...
+ `Data Source Name (DSN)`_. it nativly supports blocking, as weel as sharing
+ locks.
+
+    use Symfony\Component\Lock\Store\PostgreSqlStore;
+
+    // a PDO, a Doctrine DBAL connection or DSN for lazy connecting through PDO
+    $databaseConnectionOrDSN = 'postgresql://myuser:mypassword@localhost:5634/lock';
+    $store = new PostgreSqlStore($databaseConnectionOrDSN);
+
+In opposite to the `PdoStore`, the `PostgreSqlStore` does not need a table to stores
```
Comment (javiereguiluz) [inline components/lock.rst]: Please, use double backticks to enclose code:

```rst
[...] `PdoStore`, the `PostgreSqlStore` does [...]

-->

[...] ``PdoStore``, the ``PostgreSqlStore`` does [...]
```

**Review by javiereguiluz — APPROVED** [review]
Nice! Thanks for contributing.

I added some minor changes about some "cosmetic changes" ... but we can do those while merging if you prefer.

File: components/lock.rst (hunk 1 above)
Comment (jderusse) [reply components/lock.rst]: good catch

File: components/lock.rst (hunk 5 above)
Comment (jderusse) [reply components/lock.rst]: thanks

**Comment by javiereguiluz** [comment]
Superb contribution as always! Thanks Jérémy!

**Pull request merged**